recap/types.py

```diff
@@ -471,6 +486,9 @@ def from_dict(
                     [from_dict(t, registry) for t in type_dict.pop("types")],
                     **type_dict,
                 )
+            case "unknown":
```

Comment:
This is the part I'd strongly prefer to avoid. I am really nervous about adding `"unknown"` as a type to the CST.